.readthedocs.yaml

```diff
@@ -0,0 +1,20 @@
+# .readthedocs.yaml
+# Read the Docs configuration file
+# See https://docs.readthedocs.io/en/stable/config-file/v2.html for details
+
+version: 2
+
+build:
+  os: ubuntu-20.04
+  tools:
+    python: "3.8"
+  apt_packages:
+    - make
+    - doxygen
```

Comment:
The Doxygen package in Ubuntu 20.04 gave unfounded warnings (mqtt got warnings because the preprocessor directives in Doxyfile had no effect, and some simplelink platform had `/**===============...` interpreted as `<h1>` with a followup failure). Can this use Doxygen 1.9.4 just like the docker image?

Reply:
Can you please check if you can see wrongly generated doxygen at https://contiki-ng--2024.org.readthedocs.build/en/2024/_api/modules.html for those files?

The build output from RTD had the following warnings:
```
api-doc building API docs from "../doxygen"
api-doc executing "make -C ../doxygen > /dev/null"
warning: ignoring unsupported tag 'CREATE_SUBDIRS_LEVEL' at line 92, file Doxyfile
warning: ignoring unsupported tag 'PYTHON_DOCSTRING' at line 239, file Doxyfile
warning: ignoring unsupported tag 'NUM_PROC_THREADS' at line 477, file Doxyfile
warning: ignoring unsupported tag 'RESOLVE_UNNAMED_PARAMS' at line 547, file Doxyfile
warning: ignoring unsupported tag 'SHOW_HEADERFILE' at line 620, file Doxyfile
warning: ignoring unsupported tag 'WARN_IF_INCOMPLETE_DOC' at line 838, file Doxyfile
warning: ignoring unsupported tag 'WARN_LINE_FORMAT' at line 878, file Doxyfile
warning: ignoring unsupported tag 'CLANG_ADD_INC_PATHS' at line 1173, file Doxyfile
warning: ignoring unsupported tag 'DOCSET_FEEDURL' at line 1403, file Doxyfile
warning: ignoring unsupported tag 'FULL_SIDEBAR' at line 1614, file Doxyfile
warning: ignoring unsupported tag 'OBFUSCATE_EMAILS' at line 1645, file Doxyfile
warning: ignoring unsupported tag 'HTML_FORMULA_FORMAT' at line 1656, file Doxyfile
warning: ignoring unsupported tag 'MATHJAX_VERSION' at line 1704, file Doxyfile
warning: ignoring unsupported tag 'DOT_UML_DETAILS' at line 2495, file Doxyfile
warning: ignoring unsupported tag 'DOT_WRAP_THRESHOLD' at line 2504, file Doxyfile
warning: ignoring unsupported tag 'DIR_GRAPH_MAX_DEPTH' at line 2577, file Doxyfile
api-doc moving "../doxygen/html" to "/home/docs/checkouts/readthedocs.org/user_builds/contiki-ng/checkouts/2024/tools/readthedocs/_build/html/_api"
build succeeded.
```

Ideally we would stick to the version provided by Read the Docs unless it generates broken documentation, as they appear to discourage overriding parts of the build.

Reply:
Yes, mqtt.h is not processed with MQTT_5 enabled so https://contiki-ng--2024.org.readthedocs.build/en/2024/_api/mqtt_8h.html is missing `prop_list` as argument to `mqtt_connect` (and a bunch of other functions).

You need to look in doxygen.log for the doxygen warnings if that is the same way as 00-documentation is set up.

Reply:
Would Doxygen 1.9.1 generate these correctly? 
That's the version in Ubuntu 22.04, which is a supported RTD build environment as well; everything else will be fine as it's just sphinx modules. 
I can easily try this change in the next commit before seeing if we can override the preinstalled system doxygen.

Reply:
Doxygen randomly breaks stuff between minor releases. I don't remember what was broken with 1.9.1, it's much better than the Ubuntu 20.04 version, but #1966 was for sure still broken with 1.9.1.

Reply:
I've got it building with 1.9.4 now, so this should be resolved.